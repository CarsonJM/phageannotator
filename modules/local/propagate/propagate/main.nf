process PROPAGATE_PROPAGATE {
    tag "$meta.id"
    label 'process_high'

    conda "conda-forge::python=3.8 bioconda::bowtie2 bioconda::samtools bioconda::pysam conda-forge::numpy conda-forge::numba"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/vrhyme:1.1.0--pyhdfd78af_1' :
        'biocontainers//vrhyme:1.1.0--pyhdfd78af_1' }"

    input:
    tuple val(meta) , path(reads)
    tuple val(meta2), path(fasta)
    tuple val(meta3), path(provirus_coords)

    output:
    tuple val(meta), path("${prefix}_propagate/${prefix}_propagate.tsv")    , emit: propagate_results
    tuple val(meta), path("${prefix}_propagate/${prefix}_propagate.log")    , emit: log
    path  "versions.yml"                                                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def group = task.ext.prefix ?: "${meta.group}"
    prefix = task.ext.prefix ?: "${meta.group}_${meta.id}"
    """
    if [[ -f ${provirus_coords} ]]
    then
        propagate.py \\
        -f ${fasta} \\
        -v ${provirus_coords} \\
        -r ${reads} \\
        -o ${prefix}_propagate \\
        -t $task.cpus \\
        $args
    else
        mkdir -p ${prefix}_propagate
        echo "No integrated proviruses identified" > ${prefix}_propagate/${prefix}_propagate.tsv
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(Propagate --version 2>&1 | sed 's/^.*SPAdes genome assembler v//; s/ .*\$//')
    END_VERSIONS
    """
}
