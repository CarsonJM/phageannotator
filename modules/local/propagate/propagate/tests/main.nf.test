nextflow_process {

    name "Test Process: PROPAGATE_PROPAGATE"
    script "../main.nf"
    process "PROPAGATE_PROPAGATE"

    tag "modules"
    tag "modules_local"
    tag "propagate_propagate"


    test("fastq.gz + fasta + provirus_coords.tsv") {

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        file("$projectDir/assets/test-datasets/contigs.fasta.gz", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id: 'test' ],
                        file("$projectDir/assets/test-datasets/genomad_summary.tsv", checkIfExists: true)
                    ]
                )
                input[2] = Channel.of(
                    [
                        [ id: 'test' ],
                        file("$projectDir/assets/test-datasets/checkv_contamination.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("contigs.fasta + genomad_virus_summary + check_contamination_summary - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        file("$projectDir/assets/test-datasets/contigs.fasta.gz", checkIfExists: true)
                    ]
                )
                input[1] = Channel.of(
                    [
                        [ id: 'test' ],
                        file("$projectDir/assets/test-datasets/genomad_summary.tsv", checkIfExists: true)
                    ]
                )
                input[2] = Channel.of(
                    [
                        [ id: 'test' ],
                        file("$projectDir/assets/test-datasets/checkv_contamination.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
