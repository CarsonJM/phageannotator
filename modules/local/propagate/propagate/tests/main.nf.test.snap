{
    "fastq.gz + fasta + provirus_coords.tsv": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_propagate.tsv:md5,f5dc748967a1840bff9c504804d4b6d4"
                ]
            ],
            [
                "Time (min) |  Log                                                   ",
                "--------------------------------------------------------------------",
                "0.0         Reads input identified, using 2 threads to run Bowtie2.",
                "0.0         Checking if BAM file needs to be sorted",
                "0.0         Generating a list of all prophage regions",
                "0.0         Number of prophage regions identified: 4",
                "0.0         Number of unique host regions identified: 3",
                "0.0         Extracting coverage and performing statistical analyses",
                "0.1         Analysis finished",
                "",
                "",
                "Results file:      test_propagate.tsv",
                "Active prophages:  0",
                "",
                "                                                               ##",
                "                                                             ##  ##",
                "                                                           ##      ##",
                "######   ##  ##     ##     #######   ######    #####       ##      ##",
                "##  ##   ##  ##   ##  ##   ##        ##       ##             ##  ##",
                "######   ######   ######   ##  ###   ######    ###             ##",
                "##       ##  ##   ##  ##   ##   ##   ##           ##           ##",
                "##       ##  ##   ##  ##   #######   ######   #####            ##",
                "                                                            #  ##  #",
                "                                                           # # ## # #",
                "                                                          #   #  #   #",
                "                                                         #            #",
                ""
            ],
            [
                "versions.yml:md5,dfe5c94afe825f0011160109ed5f8bb1"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T09:52:54.460659476"
    },
    "fastq.gz + fasta + provirus_coords.tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,dfe5c94afe825f0011160109ed5f8bb1"
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test"
                        },
                        "test_propagate.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dfe5c94afe825f0011160109ed5f8bb1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T09:53:07.554244343"
    }
}